subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -82,6 +82,19 @@ Use other collections or I/O frameworks instead of `org.gradle.util.CollectionUt
 
 Gradle plugin authors may find the Designing Gradle Plugins subsection on link:https://guides.gradle.org/designing-gradle-plugins/#restricting_the_plugin_implementation_to_gradle_s_public_api[restricting the plugin implementation to Gradle's public API] helpful.
 
+[[sec:declaring_tasks]]
+=== Declaring tasks in a build script
+
+The task API gives a build author a lot of flexibility to declare tasks in a build script.
+For optimal readability and maintainability follow these rules:
+
+* The task type should be the only key-value pair that belongs within the parentheses after the task name.
+* Tasks without a specialized type should not explicitly declare api:org.gradle.api.DefaultTask[]. The type is automatically implied.
```

Comment:
While true, I consider declaring this type to make it maintainable for readers not familiar with Gradle. I don't think this is an important point to make.

Reply:
I view it more like the difference between declaring `def foo = someMethod()` vs. `File foo`. Of course we don't want to declare all conventional things. It depends on what most users know already. Source layout is pretty standardized thanks to maven, and does not need declaration.

Reply:
I removed the statement though I'd vote for adding this information _somewhere_ in the user guide. I understand the point you are making but I personally would hate to see `task myTask(type: DefaultTask)` in any code base, internal or external.